### PR TITLE
ci: autopublish rust crates on tag event

### PR DESCRIPTION
**Pull request by vincenzopalazzo**
This includes the auto-publish the release on the crate-io
and also include a rust fix (see commit for details)

I do not know how @cdecker feals about dry-run when we merge a PR
this allow us to make sure that we do not fails on release
but it can be challenging during development.

Happy to drop it if we are not sure if this is really useful

### CODE REVIEW

File: .github/workflows/crate-io.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
+on:
+  release:
+    types: [created]
+    tags:
```
Comment (cdecker) [inline .github/workflows/crate-io.yml:5]: We probably don't want to trigger on v-tags. Those are reserved for the CLN releases itself, and we are doing our best to make versions independent (multi-version schema migrations, etc)

Comment (cdecker) [reply .github/workflows/crate-io.yml:5]: Ideally we'd want to compare the published version with the version in the Cargo.toml and trigger a run if the version was increased.

**Review by cdecker — CHANGES_REQUESTED** [review]

File: .github/workflows/crate-io.yml (hunk 1 above)
Comment (vincenzopalazzo) [reply .github/workflows/crate-io.yml]: I used the `ignore-unpublished-changes: true` so if the version is already published the CI is still happy.

Do you think that this does the job that you want?

Comment (cdecker) [reply .github/workflows/crate-io.yml]: The problem is we're still compiling and testing the entire crate only to then try to upload and fail. Wasteful, but I guess quite alright :+1: We can always amend later.

**Review by cdecker — APPROVED** [review]

File: .github/workflows/crate-io.yml (hunk 1 above)
Comment (vincenzopalazzo) [reply .github/workflows/crate-io.yml]: >The problem is we're still compiling and testing the entire crate only to then try to upload and fail.

Ah yes this is true

**Pull request merged**